### ISSUE TITLE
Implement user authentication for trust metadata

### ISSUE THREAD

**Issue by AMWatkins314** [conda/conda#11921]
### Checklist

- [X] I added a descriptive title
- [X] I searched open requests and couldn't find a duplicate

### What is the idea?

Conda does not support authentication for the signature verification files `1.root.json` or `key_mgr.json` which are pointed to by the URL `signing_metadata_url_base` (as defined in condarc)
See: 
- https://github.com/conda/conda/blob/main/conda/trust/signature_verification.py#L109-L114
`# TODO: support fetching root data with credentials`

- https://github.com/conda/conda/blob/main/conda/trust/signature_verification.py#L144-L148

- https://github.com/conda/conda/blob/main/conda/trust/signature_verification.py#L187-L201
  ```
  # TODO: Figure how to handle authn for obtaining trust metadata,
  # independently of the authn used to access package repositories.
  ```

### Why is this needed?

The condarc file allows users to specify where the metadata files live via `signing_metadata_url_base`.
The `signing_metadata_url_base` can point to a location that requires user authentication.
An example of this is a repository configured to require user authentication, like Nexus or Artifactory.

### What should happen?

The user configures`signing_metadata_url_base` to point to a URL requiring user authentication. The user authentication can either be specified in a netrc file or as basic auth embedded in the URL (https://user:pwd@fqdn). When the user runs `conda install` or `conda create` the user authentication processes, the user is granted access, and packages that would be installed show the correct package signature verification information.

**Comment by awwad** [comment]
There are two reasons I know of to work on this:
- This is currently affecting users of conda / repository infrastructure who require that all contact with the server include user authentication.
- Consolidation of http request code / file download code in conda seems like a worthwhile goal.  (There are probably other issues with having duplicated `requests` calls.)

**Comment by jezdez** [comment]
This is certainly something achievable given conda's network I/O gateway system and our currently on-going work to make this pluggable (e.g. to be extended with additional authentication schemes). I would consider the fetching of content trust keys to be a specific application of the problem here.

**Comment by jezdez** [comment]
@AMWatkins314 Apologies for the long pause in communication, we've implemented the fix in #12889 but sadly missed the merge window for the July release of conda. The next _planned_ release is in September.

**Comment by jezdez** [comment]
@AMWatkins314 FYI this'll be fixed in the September release cycle in ~1 month.
